cs.c

```diff
@@ -1093,6 +1093,83 @@ cs_err CAPSTONE_API cs_option(csh ud, cs_opt_type type, size_t value)
 	return arch_configs[handle->arch].arch_option(handle, type, value);
 }
 
+CAPSTONE_EXPORT
+cs_buffer * CAPSTONE_API cs_buffer_new(size_t capacity) {
+	cs_buffer *buffer = cs_mem_malloc(sizeof(cs_buffer));
+	if (!buffer) {
+		return NULL;
+	}
+	buffer->count = 0;
+	buffer->capacity = capacity ? capacity : 64;
+	buffer->insn = cs_mem_calloc(sizeof(cs_insn), buffer->capacity);
+	if (!buffer->insn) {
+		cs_mem_free(buffer);
+		return NULL;
+	}
+	return buffer;
+}
+
+CAPSTONE_EXPORT
+void CAPSTONE_API cs_buffer_free(cs_buffer *buffer) {
+	if (buffer->insn) {
+		for (size_t i = 0; i < buffer->capacity; ++i) {
+			cs_mem_free(buffer->insn[i].detail);
+		}
+		cs_mem_free(buffer->insn);
+	}
+	cs_mem_free(buffer);
+}
+
+CAPSTONE_EXPORT
+void CAPSTONE_API cs_buffer_clear(cs_buffer *buffer) {
+	buffer->count = 0;
+}
+
+CAPSTONE_EXPORT
+bool CAPSTONE_API cs_buffer_reserve_exact(cs_buffer *buffer, size_t required) {
+	if (required == buffer->capacity) {
+		return true;
+	}
+
+	if (required < buffer->capacity) {
+		for (size_t i = required; i < buffer->capacity; ++i) {
+			if (buffer->insn[i].detail) {
+				cs_mem_free(buffer->insn[i].detail);
+			}
+		}
+	}
+	buffer->insn = cs_mem_realloc(buffer->insn, required * sizeof(cs_insn));
+	if (!buffer->insn) {
+		buffer->capacity = 0;
+		buffer->count = 0;
+		return false;
+	}
+	if (required > buffer->capacity) {
+		size_t diff = required - buffer->capacity;
+		memset(&buffer->insn[buffer->capacity], 0, diff * sizeof(cs_insn));
+	}
+
+	buffer->capacity = required;
+	if (buffer->capacity < buffer->count) {
+		buffer->count = buffer->capacity;
+	}
+
+	return true;
+}
+
+CAPSTONE_EXPORT
+bool CAPSTONE_API cs_buffer_reserve(cs_buffer *buffer, size_t additional) {
```

Comment:
~~All these API functions still documentation please. But better wait with it until the other maintainers took a look at it.~~
Haven't seen the header files. All good.